packages/components/pie-modal/src/index.ts

```diff
@@ -63,6 +63,13 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
     @validPropertyValues(componentSelector, sizes, 'medium')
     public size: ModalProps['size'] = 'medium';
 
+    @property()
+    public leadingAction!: {
```

Comment:
What does the `!` do here (sorry for the noob questions!)

Reply:
NPs. 

The `!` is a typescript thing. What it does is forces a property to always have a value. By default TS makes properties optional so they fall back to `undefined` or `null`. By adding the `!` it forces the compiler to think that the property will always have a non-null value (I think of it as filtering/excluding out the null values)

Reply:
Makes sense – does this mean that `leadingAction` must always have a value then? Or just that it should always receive an object (whether empty or with values).

Just wondering how this works for when we don't want to display a `leadingAction`

Reply:
So for displaying the `leadingAction` if it isn't provided then nothing renders. The `!` is only specific to the TS compiler itself :)

Reply:
Does it actually force it to have a value or just tell TS that it should? Do we get console errors if there is no value for `leadingAction`?

Reply:
Just TS. We won't get console errors if it isn't supplied.